examples/datadog-sink/main.tf

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  version = "~> 2.0"
+}
+   
+locals {
+  datadog_svc = element(google_service_account.datadog-viewer.*.email,0)
+  log_writ = module.log_export.writer_identity
+}
+   
+resource "google_service_account" "datadog-viewer" {
+  account_id = "${var.project_id}-datadog-viewer"
+  description = "Service account for Datadog monitoring"
+  project = var.project_id
+}
+   
+resource "google_project_iam_member" "compute-viewer" {
+  project = var.project_id
+  role    = "roles/compute.viewer"
+  member  = "serviceAccount:${var.project_id}-datadog-viewer@${var.project_id}.iam.gserviceaccount.com"
+}
+   
+resource "google_project_iam_member" "cloudasset-viewer" {
+  project = var.project_id
+  role    = "roles/cloudasset.viewer"
+  member  = "serviceAccount:${var.project_id}-datadog-viewer@${var.project_id}.iam.gserviceaccount.com"
+}
+   
+resource "google_project_iam_member" "monitoring-viewer" {
+  project = var.project_id
+  role    = "roles/monitoring.viewer"
+  member  = "serviceAccount:${var.project_id}-datadog-viewer@${var.project_id}.iam.gserviceaccount.com"
+}
+  
+resource "google_pubsub_topic_iam_member" "pubsub_sink_member" {
```

Comment:
The module should already handle this: https://github.com/terraform-google-modules/terraform-google-log-export/blob/master/modules/pubsub/main.tf#L54

Reply:
I thought so too, but this was not working during any test I ran and I couldn't figure out exactly why -- my best guess is that maybe it needs a sleep?

Reply:
Hmm, can you check again? A sleep might help but as far as I can tell this is creating the exact same resource.

Maybe share your `terraform plan` where it failed?

Reply:
Ok seems like it's working now.  Not sure what was happening before but it's entirely possible that I was doing something wrong.